test/helpers/merkleTree.js

```diff
@@ -113,7 +113,7 @@ class MerkleTree {
 
   bufDedup (elements) {
     return elements.filter((el, idx) => {
-      return this.bufIndexOf(el, elements) === idx;
+      return idx > 0 && !elements[idx - 1].equals(el);
```

Comment:
```suggestion
      return idx === 0 || !elements[idx - 1].equals(el);
```
Unless I'm mistaken, the proposed implementation always drops the first element, as opposed to keeping it.

Reply:
thanks for catching it! You are right.